### PR TITLE
Marksdoc

### PR DESCRIPTION
**Pull request by MarkR42**
Hi,

Here are some documentation updates / fixes I have made. I've also documented the XMLHttpRequest issue which I discovered.

### CODE REVIEW

File: docs/api/webserver.rst [hunk 1]
```diff
@@ -323,3 +323,9 @@ Indicate that the response is complete, and the connection with the browser is c
 is not possible to send other data (header or content) after this method.
 
 
+Webserver limitations
+=====================
+
+The webserver is based on the Mozilla built-in "jshttp" webserver used for testing. It has the following limitations:
+
+* Only one request may be processed at once. You must call response.write() and response.close() before the next request will be processed.
```
Comment (laurentj) [inline docs/api/webserver.rst]: Are you sure of this limitation? The web server supports asynchronous response for example.

**Comment by MarkR42** [comment]
Yes, I'm sure about that.

What I did was, do the "obvious" thing, of writing a slimerjs program which, in response to a HTTP request, creates its own webpage then waits for it to load, before sending the HTTP response (e.g. a jpeg screenshot).

That works just fine - except, if you send multiple requests, they get queued not executed in parallel.

Try it yourself. The jshttp seems to have this limitation. 

I also tried creating multiple servers on several different port numbers. This also doesn't allow parallel requests - again, it queues multiple requests.

**Comment by laurentj** [comment]
Ok  I will verify that.

Thank you very much for the documentation of the fs module !

**Comment by MarkR42** [comment]
Actually, I am NOT sure about it.

I wrote another test program, a simpler one, and I can't reproduce it. I suspect that maybe the requests were serialised because of something else (not the webserver).

Something definitely seems to happen, but it's not what I described. I think we should get rid of that comment. Sorry.

**Pull request merged**